mteb/evaluation/evaluators/Audio/ClassificationEvaluator.py

```diff
@@ -0,0 +1,396 @@
+        task_name: str | None = None,
+        k: int = 1,
+        encode_kwargs: dict[str, Any] = {},
+        limit: int | None = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        if limit is not None:
+            dataset_train = dataset_train.select(list(range(limit)))
+
+        self.dataset_train = AudioDataset(
+            dataset_train, audio_column_name=audio_column_name, transform=None
+        )
+        self.y_train = dataset_train[label_column_name]
+
+        self.dataset_test = AudioDataset(
+            dataset_test, audio_column_name=audio_column_name, transform=None
+        )
+        self.y_test = dataset_test[label_column_name]
+        self.task_name = task_name
+        self.encode_kwargs = encode_kwargs
+
+        if "batch_size" not in self.encode_kwargs:
+            self.encode_kwargs["batch_size"] = 32
+
+        self.k = k
+
+    def __call__(self, model, test_cache=None):
+        scores = {}
+        max_accuracy = 0
+        max_f1 = 0
+        max_ap = 0
+        dataloader_train = DataLoader(
+            self.dataset_train,
+            batch_size=self.encode_kwargs["batch_size"],
+            shuffle=False,
+            collate_fn=custom_collate_fn,
+            num_workers=min(math.floor(os.cpu_count() / 2), 16),
+        )
+        X_train = model.get_audio_embeddings(
+            dataloader_train, batch_size=self.encode_kwargs["batch_size"]
+        )
+        dataloader = DataLoader(
+            self.dataset_test,
+            batch_size=self.encode_kwargs["batch_size"],
+            shuffle=False,
+            num_workers=min(math.floor(os.cpu_count() / 2), 16),
+        )
+        if test_cache is None:
+            X_test = model.get_audio_embeddings(
+                dataloader, batch_size=self.encode_kwargs["batch_size"]
+            )
+            test_cache = X_test
+        else:
+            X_test = test_cache
+        for metric in ["cosine", "euclidean"]:  # TODO: "dot"
+            knn = KNeighborsClassifier(n_neighbors=self.k, n_jobs=-1, metric=metric)
+            knn.fit(X_train, self.y_train)
+            y_pred = knn.predict(X_test)
+            accuracy = accuracy_score(self.y_test, y_pred)
+            f1 = f1_score(self.y_test, y_pred, average="macro")
+            scores["accuracy_" + metric] = accuracy
+            scores["f1_" + metric] = f1
+            max_accuracy = max(max_accuracy, accuracy)
+            max_f1 = max(max_f1, f1)  # type: ignore
+            # if binary classification
+            if len(np.unique(self.y_train)) == 2:
+                ap = average_precision_score(self.y_test, y_pred)
+                scores["ap_" + metric] = ap
+                max_ap = max(max_ap, ap)
+        scores["accuracy"] = max_accuracy
+        scores["f1"] = max_f1
+        if len(np.unique(self.y_train)) == 2:
+            scores["ap"] = max_ap
+        return scores, test_cache
+
+
+class AudiokNNClassificationEvaluatorPytorch(Evaluator):
```

Comment:
I think it's okay to only keep the logreg one, and remove the other evaluators (same in tests).